plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/LabelValueValidator.java

```diff
@@ -26,24 +27,34 @@
 	public static final int LABEL_MAXLENGTH = 63;
 	private static final Pattern LABEL_REGEXP = Pattern.compile("^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$");
 
-	private static final String failureMessage = "A valid label value is an alphanumeric (a-z, and 0-9) string, with a maximum length of 63 characters, "
-			+ "with the '-' character allowed anywhere except the first or last character.";
+	private final String failureMessage = "A valid {0} is alphanumeric (a-z, and 0-9), "
+			+ "with the characters '-', '.' allowed anywhere except first or last.";
 
-	protected IStatus FAILED;
+	private final IStatus FAILED;
 	
 	public LabelValueValidator() {
-		FAILED = ValidationStatus.error(failureMessage);
+		this("label value");
 	}
 	
+	public LabelValueValidator(String element) {
+		FAILED = ValidationStatus.error(NLS.bind(failureMessage, element));
+	}
+
 	@Override
 	public IStatus validate(Object paramObject) {
 		if(!(paramObject instanceof String))
-			return FAILED;
+			return ValidationStatus.error("Value is not an instance of a string");
 		String value= (String) paramObject;
 		if(StringUtils.isEmpty(value))
 			return FAILED;
```

Comment:
so far in the tooling we treat the case where we have a missing required parameter as IStatus.CANCEL causing the "*" to appear instead of the error marker:
ValidationStatus.cancel(FAILED)